turbinia/workers/binary_extractor.py

```diff
@@ -41,21 +43,44 @@ def run(self, evidence, result):
     binary_extraction_evidence = BinaryExtraction()
 
     binary_extraction_evidence.local_path = self.output_dir
-    image_export_log = os.path.join(
-        self.output_dir, '{0:s}.log'.format(self.id))
+    image_export_log = os.path.join(self.output_dir, 'binary_extraction.log')
+
+    binary_extraction_dir = os.path.join(self.output_dir, 'extracted_binaries')
 
     cmd = [
-        'image_export.py', '--signatures', 'elf,exe_mz', '--logfile',
-        image_export_log
+        'image_export.py', '--partitions', 'all', '--no_vss', '--signatures',
+        'elf,exe_mz', '--logfile', image_export_log
     ]
     if config.DEBUG_TASKS:
       cmd.append('-d')
-    cmd.extend(['-w', self.output_dir, evidence.local_path])
+    cmd.extend(['-w', binary_extraction_dir, evidence.local_path])
 
     result.log('Running image_export as [{0:s}]'.format(' '.join(cmd)))
 
     self.execute(
         cmd, result, log_files=[image_export_log],
         new_evidence=[binary_extraction_evidence], close=True)
 
+    json_path = os.path.join(
+        self.output_dir, 'extracted_binaries', 'hashes.json')
+
+    with open(json_path) as json_file:
+      hashes = json.load(json_file)
+
+    files = next(os.walk(binary_extraction_dir))[2]
```

Comment:
I don't quite understand this.  Won't the output of `image_export.py` have multiple directories that match the original on disk paths?  If so, you'll need to iterate over the output in order to get all the filenames?  If not and you only want output for one dir, you can also just use `os.listdir()`.

Reply:
Yes, image_will maintain original folder structure. In order to get number of files extracted I need to recursively traverse through all directories, not just the root. That's why I have to use os.walk(), since os.listdir() will just list the root directory. Below is a tree output of a sample extraction. 

. (/extracted_binaries)
├── bin
│   ├── bzcat
│   └── echo
├── sbin
│   ├── visudo
│   └── tune2fs
└── hashes.json

Reply:
Actually, I realized that I used os.walk in a wrong way, please see updated code.